contrib/mesos/pkg/minion/server.go

```diff
@@ -243,6 +252,22 @@ func (ms *MinionServer) Run(hks hyperkube.Interface, _ []string) error {
 		log.Fatalf("No API client: %v", err)
 	}
 	ms.clientConfig = clientConfig
+	c := client.NewOrDie(clientConfig)
+
+	// create or update node object on apiserver. This must be done synchronously
+	// because the kubelet will refuse launching pods if the labels do not match.
+	labels := map[string]string{}
+	if ms.nodeLabelString != "" {
+		err = json.Unmarshal([]byte(ms.nodeLabelString), &labels)
+		if err != nil {
+			log.Fatalf("Invalid node-labels string: %v", err)
+		}
+	}
+	log.Infof("Creating or updating node %s with labels %v", ms.nodeHostName, labels)
+	_, err = node.CreateOrUpdate(c, ms.nodeHostName, labels)
```

Comment:
Complexity. I prefer KISS.

We should split the executor and the kubelet mid-term completely into their own processes. For simplicity I want to minimize any dependencies between the two. They should start-up and just find each other. If one dies and is restarted, they should re-find each other.